tasks/request_files/request_files.py

```diff
@@ -235,263 +245,210 @@ def inner_task(event: Dict, max_retries: int, retry_sleep_secs: float,
-                        None,
-                        RequestMethod.POST,
-                        db_queue_url,
-                        max_retries,
-                        retry_sleep_secs)
-
+                        restore_expire_days,
+                        glacier_bucket,
+                        attempt,
+                        job_id,
+                        retrieval_type,
+                    )
+                    a_file[FILE_SUCCESS_KEY] = True
+                    a_file[FILE_ERROR_MESSAGE_KEY] = ""
                 except ClientError as err:
                     LOGGER.warning(err)
                     a_file[FILE_ERROR_MESSAGE_KEY] = str(err)
 
         attempt = attempt + 1
-        if attempt <= max_retries + 1:  # Only sleep if not on last attempt. # todo: Use backoff code. ORCA-201
+        if (
+            attempt <= max_retries + 1
+        ):  # Only sleep if not on last attempt. # todo: Use backoff code. ORCA-201
             # Check for early completion.
-            if all(a_file[FILE_SUCCESS_KEY] for a_file in granule[GRANULE_RECOVER_FILES_KEY]):
+            if all(
+                a_file[FILE_SUCCESS_KEY]
+                for a_file in granule[GRANULE_RECOVER_FILES_KEY]
+            ):
                 break
             time.sleep(retry_sleep_secs)
 
+    files = []
     any_error = False
     for a_file in granule[GRANULE_RECOVER_FILES_KEY]:
         # if any file failed, the whole granule will fail
         if not a_file[FILE_SUCCESS_KEY]:
             any_error = True
-            # If this is reached, that means there is no entry in the db for file's status.
-            post_status_for_file_to_queue(
-                job_id, granule_id, os.path.basename(a_file[FILE_KEY_KEY]),
-                a_file[FILE_KEY_KEY],
-                a_file[FILE_DEST_BUCKET_KEY],
-                ORCA_STATUS_FAILED,
-                a_file.get(FILE_ERROR_MESSAGE_KEY, None),
-                request_time,
-                datetime.now(timezone.utc).isoformat(),
-                None,
-                RequestMethod.POST,
-                db_queue_url,
-                max_retries,
-                retry_sleep_secs)
+        
+        files.append(
+            {
+                "filename": os.path.basename(a_file[FILE_KEY_KEY]),
+                "key_path": a_file[FILE_KEY_KEY],
+                "restore_destination": a_file[FILE_DEST_BUCKET_KEY],
+                "status_id": shared_recovery.OrcaStatus.FAILED.value if not a_file[FILE_SUCCESS_KEY] else shared_recovery.OrcaStatus.PENDING.value,
+                "error_message": a_file[FILE_ERROR_MESSAGE_KEY] if not a_file[FILE_SUCCESS_KEY] else None,
+                "request_time": datetime.now(timezone.utc).isoformat(),
+                "last_update": datetime.now(timezone.utc).isoformat(),
+                "completion_time": datetime.now(timezone.utc).isoformat() if not a_file[FILE_SUCCESS_KEY] else None
+            }
+        )
```

Comment:
This will work, you may want to consider consolidating the logic to make it a bit cleaner. As an example, you could do something like this as well.

```
    files = []
    any_error = False
    for a_file in granule[GRANULE_RECOVER_FILES_KEY]:
        # Create the dictionary for a "successful granule"
        file = {
                "filename": os.path.basename(a_file[FILE_KEY_KEY]),
                "key_path": a_file[FILE_KEY_KEY],
                "restore_destination": a_file[FILE_DEST_BUCKET_KEY],
                "status_id": shared_recovery.OrcaStatus.PENDING.value,
                "error_message": None,
                "request_time": datetime.now(timezone.utc).isoformat(),
                "last_update": datetime.now(timezone.utc).isoformat(),
                "completion_time": None
            }

        # if any file failed, the whole granule will fail and the file information should be updated
        if not a_file[FILE_SUCCESS_KEY]:
            any_error = True
            file["status_id"] = shared_recovery.OrcaStatus.FAILED.value
            file["error_message"] = a_file[FILE_ERROR_MESSAGE_KEY]
            file["completion_time" = datetime.now(timezone.utc).isoformat()

        files.append(file)
```

Reply:
works for me.